back/babble/src/main/java/gg/babble/babble/domain/user/User.java

```diff
@@ -50,6 +54,12 @@ public User(final Long id, final String nickname, final Room room) {
         this.id = id;
         this.nickname = nickname;
         this.room = room;
+        this.avatar = avatarByNickname(nickname);
+    }
+
+    public static String avatarByNickname(final String nickname) {
+        long avatarIndex = ((long) nickname.hashCode() - Integer.MIN_VALUE) % NUMBER_OF_AVATAR;
+        return String.format(AVATAR_FORMAT, avatarIndex);
```

Comment:
PR에 적은 코멘트도 보았고, 의도도 파악했습니다.  


한가지 궁금한점은 
`(long) nickname.hashCode() + Integer.MAX_VALUE` 가 아닌  
`(long) nickname.hashCode() - Integer.MIN_VALUE` 로 하신 이유가 있을까요?

전자가, nickname.hashCode() 에서 **음수가 나온 경우를 대비한다**라는 메시지를 더 잘 보여준다고 생각이 드네요.  
후자는 개인적으로 한눈에 들어오진 않았어요.

그리고 제가 코드를 짯다면 음수가 나오는 경우를 방지하기위해서

```suggestion
    public static String avatarByNickname(final String nickname) {
        int avatarIndex = Math.abs(nickname.hashCode()) % NUMBER_OF_AVATAR;
        return String.format(AVATAR_FORMAT, avatarIndex);
```

- 이렇게 짯을꺼 같긴하네요 ㅋ_ㅋ

Reply:
전자의 방법은 MAX_INT와 MIN_INT가 달라서 `-1`인 경우가 생길거에요!

아래 방법은 가능하겠네요!

Reply:
아 음수가 한개 더 많으니, -1 인 경우가 있을수 있겠네요. ok

Reply:
참고로 여담이지만 long 타입 연산은 int 보다는 생각보다 시간이 꽤 많이걸립니다.  

다른말로하면 int가 특별하게 빨라요. jvm은 4byte 크기를 기본적으로 처리하거든요.   
jvm 말고 다른 언어도 마찬가지! 그래서 다른 타입보다 일반적으로 int를 많이 사용합니다.  

예를들어 char의경우 1byte임에도 불구하고 추가연산이 필요해져서 int보다 느립니다. 어셈블리 까보시면 바로 알수 있어요 ㅎㅎ. 

ps의 경우에도 어떤 문제들은 int를 안썻다가 시간초과(TLE)가 터지는경우가 가끔 있어요. ㅋㅋ